src/Nethermind/Nethermind.Init/Steps/EthereumStepsManager.cs

```diff
@@ -153,6 +153,7 @@ private void RunOneRoundOfInitialization(CancellationToken cancellationToken)
                     {
                         if (_logger.IsWarn) _logger.Warn(
                             $"Step {step.GetType().Name.PadRight(24)} failed after {stopwatch.ElapsedMilliseconds}ms");
+                        throw t.Exception;
```

Comment:
You know you are resetting call stack here?
Does this throw on some pool thread?

Reply:
Not entirely sure if will work but try `ExceptionDispatchInfo.Capture(t.Exception).Throw();`